File: app/src/main/java/com/example/theduckcardchatsystem/viewmodel/AddContactsViewModel.java
package com.example.theduckcardchatsystem.viewmodel;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.recyclerview.widget.RecyclerView;

import com.example.theduckcardchatsystem.repository.AddContactsRepository;

public class AddContactsViewModel extends AndroidViewModel {
    private Application application;
    private AddContactsRepository addContactsRepository;
    public AddContactsViewModel(@NonNull Application application) {
        super(application);
        this.application = application;
        addContactsRepository = new AddContactsRepository(application);

    }
    public void initRecyclerView(RecyclerView recyclerView, Context context){
        addContactsRepository.initRecyclerView(recyclerView, context);
    }
}

File: app/src/main/java/com/example/theduckcardchatsystem/viewmodel/AddContactsViewModel.kt
package com.example.theduckcardchatsystem.viewmodel

import android.app.Application
import android.content.Context
import androidx.lifecycle.AndroidViewModel
import com.example.theduckcardchatsystem.repository.AddContactsRepository
import androidx.recyclerview.widget.RecyclerView
import com.example.theduckcardchatsystem.repository.ChatsRepository
import com.example.theduckcardchatsystem.repository.ContactsRepository
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.example.theduckcardchatsystem.ui.model.CommonModel
import com.example.theduckcardchatsystem.repository.ContactsRepository.ContactsHolder
import com.example.theduckcardchatsystem.repository.EnterCodeRepository
import com.example.theduckcardchatsystem.ui.login.LoginActivity
import com.example.theduckcardchatsystem.repository.EnterPhoneRepository
import com.example.theduckcardchatsystem.repository.GroupChatRepository

class AddContactsViewModel(private val application: Application) : AndroidViewModel(
    application
) {
    private val addContactsRepository: AddContactsRepository
    fun initRecyclerView(recyclerView: RecyclerView?, context: Context?) {
        addContactsRepository.initRecyclerView(recyclerView, context)
    }

    init {
        addContactsRepository = AddContactsRepository(application)
    }
}